src/frontend/Canonicalize.ml

```diff
@@ -159,12 +167,21 @@ let rec parens_stmt {stmt; smeta} =
           ; identifier
           ; initial_value= Option.map ~f:no_parens init
           ; is_global }
+    | While (e, s) -> While (e, parens_stmt (stmt_to_block s))
+    | IfThenElse (e, s1, s2) -> (
+      match s2 with
+      | Some x ->
+          IfThenElse
+            ( e
```

Comment:
I think we need to propagate the parens normalizing on this expression. Notice the extra parenthesis on one of the unit tests, shouldn’t happen

Reply:
Do you mean `no_parens e`? This seems to solve the extra parens on the unit test below.